piptools/resolver.py

```diff
@@ -648,7 +648,7 @@ def _do_resolve(
 
             # Collect all incompatible install requirement names
             cause_ireq_names = {
-                key_from_req(cause.requirement) for cause in cause_exc.causes
+                key_no_extra_from_req(cause.requirement) for cause in cause_exc.causes
```

Comment:
Since this is the only place where the new function is used, could you probably merge it with `key_from_req` creating an optional `strip_extras` parameter?

I don't think that would differ with other usages.

Reply:
In my own projects, I am somewhat wary of such "flag" parameters. But sure, no problem.